### ISSUE TITLE
Mockito extension is ignored when running with Spek

### ISSUE THREAD

**Issue by vokilam** [raniejade/spek-idea-plugin#36]
Hi.
There is an [incubating feature](https://github.com/mockito/mockito/wiki/What's-new-in-Mockito-2#mock-the-unmockable-opt-in-mocking-of-final-classesmethods) of mockito framework that allows to mock previously unmockable entities like final classes.

Here is a sample project that demonstrates the issue: https://github.com/vokilam/test-spec-mockito/tree/master/app/src/test/java/com/vokilam/test_spec_mockito.

The problem is that this feature is ignored when running with Spek plugin (see `ExampleUnitTestKotlinSpek.kt`). Both java junit test (`ExampleUnitTest.java`) and kotlin junit test (`ExampleUnitTestKotlin.kt`) works as expected (final class is mocked).

In case of Spek `test/resources/mockito-extensions/org.mockito.plugins.MockMaker` file is ignored and I got the following output:

```
Testing started at 18:56 ...
May 07, 2017 6:56:50 PM org.junit.platform.launcher.core.ServiceLoaderTestEngineRegistry loadTestEngines
INFO: Discovered TestEngines with IDs: [spek]
org.mockito.exceptions.base.MockitoException: 
Cannot mock/spy class com.vokilam.test_spec_mockito.MainActivity
Mockito cannot mock/spy because :
 - final class
```

I'm not 100% sure this problem is connected with Spek plugin, however I wanted to identify a problem and to direct it to a corresponding tracker.

**Comment by raniejade** [comment]
In your kotlin unit test please remove `@Throws(Exception::class)`, it should fail as well. Can you try adding `:assembleDebugUnitTest` as a before run task for the run configuration?

**Comment by vokilam** [comment]
Removed `@Throws` annotation – same behavior as before: kotlin unit test is passing, final class is mocked (no error messages from mockito).

Added `:assembleDebugUnitTest` task to default configuration for Spek (as described [here](https://blacklenspub.com/bacon-driven-development-in-kotlin-with-spek-376780e7ab3a#142b)). Before that I did it manually. Same behavior for spek test – mokito extension is not working.

**Comment by raniejade** [comment]
That's weird, I'll have to test with an android project. It's working with a non-android one. I'll get back to you if I have leads.

**Comment by vokilam** [comment]
My environment is 

Android Studio 2.3.1
Build #AI-162.3871768, built on April 1, 2017
JRE: 1.8.0_112-release-b06 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

macOS Sierra 10.12.4

**Comment by vokilam** [comment]
Anything I can help you with?

**Comment by raniejade** [comment]
@vokilam Sorry, I haven't really have started the investigation - I'm a bit busy with work stuff right now.

**Comment by raniejade** [comment]
@vokilam i'm setting up my android environment, by any chance is the test passing when running from the command line?

**Comment by vokilam** [comment]
Cannot check until tomorrow, sorry. Which task should I run?

**Comment by raniejade** [comment]
I'm not sure - not an android guy, probably the task that runs the unit tests :)

**Comment by vokilam** [comment]
Let me check, I will get back

**Comment by vokilam** [comment]
`./gradlew test` – completes successfully, reports are in app/build/reports/tests folder. I had to annotate test class with `@RunWith(JUnitPlatform::class)` so it can be picked by gradle.

**Comment by raniejade** [comment]
So most likely the problem is within the IDE. While using `@RunWith(JUnitPlatform::class)` is the test passing when running from the IDE? If it passes, the problem is with the spek plugin, if not most likely it's an IDE issue.

**Comment by vokilam** [comment]
I'm a bit confused with results I get. I will check in IDEA also and get back :)

**Comment by vokilam** [comment]
Okay, here are my results.

**Set up IDEA**
I set up a gradle based java project in IDEA with junit 5. 
I created a classic java test and spek test. Each test mocks a `public final class MyClass`.

    MyClass test = Mockito.mock(MyClass.class);

I created a file `test/resources/mockito-extensions/org.mockito.plugins.MockMaker` with following line (configure Mockito extension – now it should mock final classes):

    mock-maker-inline

**Results**

1.  Run `./gradlew junitPlatformTest` – two tests succeeded
2. Run java test from IDE – success
3. Run spek test from IDE (spek plugin) – success
4. Comment `mock-maker-inline` line and rerun previous tests – all failed with following exception (as expected):
    org.mockito.exceptions.base.MockitoException: 
    Cannot mock/spy class MyClass
    Mockito cannot mock/spy because: final class

**Set up Android Studio**
I set up a gradle based Android project with junit 4.
Same setup for test classes. Spek test class has `@RunWith(JUnitPlatform::class)` so it can be visible when testing from console.
Same setup for mockito extension.

**Results**

1. Run `./gradlew testDebugUnitTest` – results are in `build/reports/tests/testDebugUnitTest`, two tests succeeded
2. Run java test from IDE – success
3. Run spek test from IDE – **failed** with `MockitoException: Cannot mock/spy class MainActivity` – that was not expected
4. Comment `mock-maker-inline` line and rerun previous tests – all failed with exception.

**Set up Android Studio JUnit 5**
I set up a gradle based Android project with[ junit 5](https://github.com/aurae/android-junit5).
Same setup for test classes. `@RunWith(JUnitPlatform::class)` annotation is removed because test is already visible for console command.
Same setup for mockito extension.

1. Run `./gradlew junitPlatformTestDebug` – two tests succeeded
2. Run java test from IDE – success
3. Run spek test from IDE – **failed** with `MockitoException: Cannot mock/spy class MainActivity` – that was not expected
4. Comment `mock-maker-inline` line and rerun previous tests – all failed with exception.

Summing up: Mockito extension is not applied when test is run from Android Studio via Spek plugin (junit4 and junit5). Same setup for IDEA (junit5) works fine.

**Comment by raniejade** [comment]
Hi, sorry for the very long delay. I still cant replicate it, can you try moving `test/resources/mockito-extensions/org.mockito.plugins.MockMaker` to `main/resources/mockito-extensions/org.mockito.plugins.MockMaker` and see if it passes?

**Comment by elyashivlavi** [comment]
Hi, I'm facing the same issue...
I tried put this extension in `main/resources`, but still get the same error...
